src/client/proxy/http.cc

```diff
@@ -500,15 +503,24 @@ bool HTTPMessage::SaveJumpServiceAddress()
 {
   try
     {
+      if (m_Base64Destination.empty())
+        {
+          LOG(debug) << "HTTPProxyHandler: empty Base64 destination";
+          return false;
+        }
+      if (m_Address.empty())
+        ParseURL();
```

Comment:
By now, you shouldn't need to parse anything more than once.

Reply:
100% agree, I went overkill on this. I'll clean up this area in the HTTP proxy rewrite.